sceptre/config/reader.py

```diff
@@ -413,6 +414,16 @@ def _recursive_read(
 
         return config
 
+    def _write_debug_file(self, content):
+        with tempfile.NamedTemporaryFile(
+            mode="w", delete=False, prefix="rendered_"
+        ) as temp_file:
+            temp_file.write(content)
```

Comment:
Just my two cents, but I think writing to a temp file made some sense when the idea is that this would happen automatically. But rather than make ANY default I'd personally prefer if the user specified a path. I propose a new switch: `--dump-config <path>` dedicated to this purpose. It's usage is clear to any user and it's behavior is easy to understand.

Since it's possible we'll be rendering multiple configs, I think @zaro0508 is right about the need for a directory. So rendered configs would go to `<path>/the-stack-name.yaml`.

Reply:
i think `--dump-config <path>` can be done in a follow on PR.

@alexharv074 says errors exit on first failure so i'm fine with writing to the system temp directory